Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm

```diff
@@ -180,13 +180,16 @@
         return;
 
     auto category = AudioSession::CategoryType::None;
+    auto mode = AudioSession::Mode::Default;
     if (AudioSession::sharedSession().categoryOverride() != AudioSession::CategoryType::None)
         category = AudioSession::sharedSession().categoryOverride();
-    else if (captureCount || (isPlayingAudio && AudioSession::sharedSession().category() == AudioSession::CategoryType::PlayAndRecord))
+    else if (captureCount || (isPlayingAudio && AudioSession::sharedSession().category() == AudioSession::CategoryType::PlayAndRecord)) {
         category = AudioSession::CategoryType::PlayAndRecord;
-    else if (hasAudibleAudioOrVideoMediaType)
+        mode = AudioSession::Mode::VideoChat;
+    } else if (hasAudibleAudioOrVideoMediaType) {
         category = AudioSession::CategoryType::MediaPlayback;
-    else if (webAudioCount)
+        mode = AudioSession::Mode::MoviePlayback;
```

Comment:
We should only set `mode = AudioSession::Mode::MoviePlayback` when there's a non-zero count of `VideoAudio` sessions.